### ISSUE TITLE
[🐛 BUG] On_submission_change - submission is completed twice

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#2152]
### What went wrong? 🤔

The callback is called twice sometimes with the same parameters. It leads to notifying twice the user that the scenario is completed.

## How to replicate

```python
from taipy import Config
import taipy as tp
from taipy.gui import Gui, notify


# Normal function used by Taipy
def double(nb):
    return nb * 2


# Configuration of Data Nodes
input_cfg = Config.configure_data_node("input_dn", default_data=21)
output_cfg = Config.configure_data_node("output_dn")


# Configuration of tasks
first_task_cfg = Config.configure_task("double",
                                       double,
                                       input_cfg,
                                       output_cfg)


# Configuration of scenario
scenario_cfg = Config.configure_scenario(id="my_scenario",
                                         task_configs=[first_task_cfg],
                                         name="my_scenario")



def notify_from_submissions(state, submittable, details):
    submission_status = details.get('submission_status')

    if submission_status == 'COMPLETED':
        print("COMPLETED")
        print(submittable)
        print(details)
        notify(state, 'success', 'Completed!')
        # Add additional actions here, like updating the GUI or logging the completion.

    elif submission_status == 'FAILED':
        print("FAILED")
        notify(state, 'error', 'FAILED!')
        # Handle failure, like sending notifications or logging the error.


if __name__=="__main__":
    tp.Orchestrator().run()
    scenario_1 = tp.create_scenario(scenario_cfg)

    scenario_md = """
<|{scenario_1}|scenario|on_submission_change=notify_from_submissions|>
"""
    Gui(scenario_md).run()
```

This is what I receive sometimes. The callback is called twice:

```console
[2024-10-25 11:50:04.385][Taipy][INFO] job JOB_double_bd8e666f-5ce5-40ad-8694-af7d02a3faec is completed.
COMPLETED
COMPLETED
<taipy.core.submission.submission.Submission object at 0x000001BAC6CCCE90>
<taipy.core.submission.submission.Submission object at 0x000001BAC6CCD310>
{'submission_status': 'COMPLETED', 'submittable_entity': my_scenario, 'origin_entity_id': 'SCENARIO_my_scenario_6c3d70af-a896-46f8-a0dc-4f17628b29d4', 'origin_entity_type': 'SCENARIO', 'origin_entity_config_id': 'my_scenario', 'creation_date': datetime.datetime(2024, 10, 25, 11, 50, 4, 207191), 'version': '5395d6e1-e5b7-4c2e-a391-7f71d04e8047', 'job_triggered_submission_status_changed': 'JOB_double_bd8e666f-5ce5-40ad-8694-af7d02a3faec'}
{'submission_status': 'COMPLETED', 'submittable_entity': my_scenario}
```

## Expected behavior

This callback should be called once with these parameters

### Version of Taipy

- release/4.0
- develop

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by toan-quach** [comment]
Hmmm I'm running it on the develop branch and it doesn't seem to have this issue
![image](https://github.com/user-attachments/assets/a0f69c6c-d456-437a-be4a-f5db99d87e9d)

**Comment by toan-quach** [comment]
But the problem does happen on 4.0!
![image](https://github.com/user-attachments/assets/541d8390-8418-42d4-85b3-46ab64aac632)

**Comment by jrobinAV** [comment]
I can see it is closed. 
It should be backported to release/4.0. Did I miss the PR for the backport, or is it planned for the future?
@toan-quach @FredLL-Avaiga

**Comment by FredLL-Avaiga** [comment]
I guess @FabienLelaquais has backported this ?

**Comment by jrobinAV** [comment]
All right. Thx!

**Comment by FlorianJacta** [comment]
It doesn't seem to be fixed in 4.0.2.dev1.

```
[2024-12-18 14:47:46.742][Taipy][INFO] Updating configuration with command-line arguments...
[2024-12-18 14:47:46.743][Taipy][INFO] Managing application's version...
[2024-12-18 14:47:46.744][Taipy][INFO] Development mode: Clean all entities of version 1b849301-77bb-43bd-b4c4-c4f0f33d15b5
[2024-12-18 14:47:46.793][Taipy][INFO] Checking application's version...
[2024-12-18 14:47:46.794][Taipy][INFO] Blocking configuration update...
[2024-12-18 14:47:46.795][Taipy][INFO] Starting job dispatcher...
[2024-12-18 14:47:46.798][Taipy][INFO] Orchestrator service has been started.
[2024-12-18 14:47:48.220][Taipy][INFO]  * Server starting on http://127.0.0.1:5000
[2024-12-18 14:47:52.302][Taipy][INFO] job JOB_double_2312965a-c72d-4649-aa26-53e9a2c74e68 is completed.
COMPLETED
<taipy.core.submission.submission.Submission object at 0x000002B99723BC10>
COMPLETED
<taipy.core.submission.submission.Submission object at 0x000002B9972600D0>
{'submission_status': 'COMPLETED', 'submittable_entity': my_scenario}
{'submission_status': 'COMPLETED', 'submittable_entity': my_scenario, 'origin_entity_id': 'SCENARIO_my_scenario_069b8ca0-845d-4929-a1bc-b59341925b2c', 'origin_entity_type': 'SCENARIO', 'origin_entity_config_id': 'my_scenario', 'creation_date': datetime.datetime(2024, 12, 18, 14, 47, 52, 88398), 'version': '1b849301-77bb-43bd-b4c4-c4f0f33d15b5', 'job_triggered_submission_status_changed': 'JOB_double_2312965a-c72d-4649-aa26-53e9a2c74e68'}
```

**Comment by FredLL-Avaiga** [comment]
on it

**Comment by AlexandreSajus** [comment]
Fixed in 4.0.2.dev2!
